vcd/resource_vcd_lb_server_pool.go

```diff
@@ -0,0 +1,344 @@
+							Computed:    true,
+							Type:        schema.TypeString,
+							Description: "Pool member id (formatted as member-xx, where xx is a number)",
+						},
+						"condition": &schema.Schema{
+							Type:         schema.TypeString,
+							ForceNew:     false,
+							Required:     true,
+							ValidateFunc: validateCase("lower"),
+							Description:  "Defines member state. One of enabled, drain, disabled.",
+						},
+						"name": {
+							Required:    true,
+							ForceNew:    false,
+							Type:        schema.TypeString,
+							Description: "Name of pool member",
+						},
+						"ip_address": {
+							Optional:    true,
+							ForceNew:    false,
+							Type:        schema.TypeString,
+							Description: "IP address of member in server pool",
+						},
+						"port": {
+							Required:    true,
+							ForceNew:    false,
+							Type:        schema.TypeInt,
+							Description: "Port at which the member is to receive traffic from the load balancer",
+						},
+						"monitor_port": {
+							Required:    true,
+							ForceNew:    false,
+							Type:        schema.TypeInt,
+							Description: "Port at which the member is to receive health monitor requests. Can be the same as port",
+						},
+						"weight": {
+							Required:    true,
+							ForceNew:    false,
+							Type:        schema.TypeInt,
+							Description: "Proportion of traffic this member is to handle. Must be an integer in the range 1-256",
+						},
+						"min_connections": {
+							Optional:    true,
+							ForceNew:    false,
+							Type:        schema.TypeInt,
+							Description: "Minimum number of concurrent connections a member must always accept",
+						},
+						"max_connections": {
+							Optional: true,
+							ForceNew: false,
+							Type:     schema.TypeInt,
+							Description: "The maximum number of concurrent connections the member can handle. If exceeded " +
+								"requests are queued and the load balancer waits for a connection to be released",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVcdLBServerPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	LBPool, err := getLBPoolType(d)
+	if err != nil {
+		return fmt.Errorf("unable to expand load balancer server pool: %s", err)
+	}
+
+	createdPool, err := edgeGateway.CreateLBServerPool(LBPool)
+	if err != nil {
+		return fmt.Errorf("error creating new load balancer server pool: %s", err)
+	}
+
+	// We store the values once again because response includes pool member IDs
+	if err := setLBPoolData(d, createdPool); err != nil {
+		return err
+	}
+	d.SetId(createdPool.ID)
+	return nil
+}
+
+func resourceVcdLBServerPoolRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Seems only minor functionality is different vs `datasourceVcdLbServerPoolRead` - possible for reuse

Reply:
Yes. Partly.
I looked at it in the previous resource (service monitor) as well, but there are a few things that differ:
* Lookup is done by name (data source) or by id (resource)
* Data source read sets the ID (resource does not)
* Resource read unsets the ID if it is not found (data source does not)

Having in mind the difference IMHO If I made some wrapper function with arguments it wouldn't save neither code lines, nor clarity.